### PR TITLE
ipv6 support: addr scope should be global or site

### PR DESCRIPTION
**Pull request by pacoxu**
## What type of PR is this?
/kind bug

## What this PR does / why we need it:

When I use fefe:ffff:1::100 as default ipv6 pool with calico， the pod ip cannot be shown in pod spec. These ipv6 addr are with site scope, not global scope. 

However, it seems that site is deprecated but still can used inside cluster. 
Since https://tools.ietf.org/html/rfc3879, deprecating Site Local Addresses, I'm not sure whether we should support site ipv6 ip.

## Additional documentation 
I tried calico dual stack mode with kubernetes and my ipv6 addr cannot be shown in pod spec.

```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
4: eth0@if27: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default
    link/ether 36:10:b4:f4:03:c0 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.29.125.4/32 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fefe:ffff::ff3b:3eea:d343/128 scope site
       valid_lft forever preferred_lft forever
    inet6 fe80::3410:b4ff:fef4:3c0/64 scope link
       valid_lft forever preferred_lft forever
```

After looking into the code, I find the ipv6 address of calico pod is scope site(in another environement, the scop is global and this one is site in ipv6 dual stack.

According to https://serverfault.com/questions/63014/ip-address-scope-parameter, I think both global and site should be accepted here. Calico
> from http://linux-ip.net/html/tools-ip-address.html :
> 
> Scope | Description
> global | valid everywhere
> site | valid only within this site (IPv6)
> link | valid only on this device
> host | valid only inside this host (machine)

``` release note
fixes of bugs which have not been released
```

Signed-off-by: pacoxu <paco.xu@daocloud.io>

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
Hi @pacoxu - thanks for your pr :)

Would you mind updating your pr description to conform to the PR template: https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md. It'll make things easier for whatever reviewer picks this up (and also certain parts of the template, like the release note, are a prerequisite for your PR being merged). Thanks!

/ok-to-test

File: pkg/kubelet/dockershim/network/plugins.go [hunk 1]
```diff
@@ -242,16 +242,23 @@ func getOnePodIP(execer utilexec.Interface, nsenterPath, netnsPath, interfaceNam
...
-	}
-	ip, _, err := net.ParseCIDR(fields[3])
-	if err != nil {
-		return nil, fmt.Errorf("CNI failed to parse ip from output %s due to %v", output, err)
+	for  _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) < 6 {
+			continue
+		}
+		if flieds[5] != "global" && flieds[5] != "site" {
```
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/network/plugins.go]: Nit: I'm fairly confident tests are going to fail bc of a typo here :) (i.e. `s/flieds/fields/`).

**Comment by aojea** [comment]
Site local addresses are the ones that belong to the range `FEC0::/10` and are deprecated as it was pointed out in the description, that means that can stop to work without any notice or maybe some other integrations that uses the pods IP doesn't consider valid causing a compatibility problem.

We are not allowing  special addresses/ranges in IPv4 and IPv6, like multicat, link-local , unspecified , ... so I suggest to not allow site-local too.

**Comment by pacoxu** [comment]
@aojea I agree that it would be better to validate cluster cidr/ service ip range than supporting them， as site local are deprecated.

However, before site local ips are removed, we can show the site local ipv6 ips for pod. This is not contradictory

**Comment by aojea** [comment]
> However, before site local ips are removed, we can show the site local ipv6 ips for pod. This is not contradictory

My main concern is if those ips are really going to work with kube-proxy or whatever implementation they use to implement services, and doesn't seem good to allow ips that may break this functionality.

**Pull request closed without merging**